app/Dockerfile

```diff
@@ -0,0 +1,63 @@
+# from https://bun.sh/guides/ecosystem/docker, with modifications
+# to run a hot-reloading development server
+
+# use the official Bun image
+# see all versions at https://hub.docker.com/r/oven/bun/tags
+FROM oven/bun:1 AS base
+WORKDIR /app
+
+
+# -----------------------------------------------------------
+# install dependencies for dev and prod into temp directories
+# this will cache them and speed up future builds
+FROM base AS install
+
+COPY package.json bun.lockb /temp/dev/
+RUN --mount=type=cache,target=/tmp/bun \
+    cd /temp/dev/ && \
+    bun install --cache-dir /tmp/bun --frozen-lockfile
+
+# install with --production (exclude devDependencies)
+COPY package.json bun.lockb /temp/prod/
+RUN --mount=type=cache,target=/tmp/bun \
+    cd /temp/prod && \
+    bun install --cache-dir /tmp/bun --frozen-lockfile --production
+
+
+# -----------------------------------------------------------
+# copy node_modules from temp directory
+# then copy all (non-ignored) project files into the image
+FROM base AS prerelease
+COPY --from=install /temp/dev/node_modules node_modules
+COPY . .
+
+# [optional] tests & build
+ENV NODE_ENV=production
+RUN bun test
```

Comment:
This should be `bun run test`. `bun test` will use bun's own testing framework which is not what we want.

Also this probably isn't super necessary, because there's a github action that tests on every PR push.

Reply:
FWIW, a lot of that Dockerfile comes from bun's example, and frankly I think it's overcomplicated. If you're ok with it I'll just replace the multi-stage build stuff with a straightforward Dockerfile, keeping your comments about using `vite` rather than `bun` in mind.

Reply:
Oh okay, I didn't know if that was part of aligning the frontend with some specific staging/test/production process you had decided to use for the rest of the stack.

If we want the absolute minimum, maybe we just need:

```dockerfile
FROM oven/bun:1 AS base
WORKDIR /app
RUN bun install
RUN bun run dev
```

And expose and entrypoint if needed?

We could have production "section" (not sure what you call these `from as` things in docker) where it more exactly replicates the production environment. But we'll also have the netlify PR previews to test that, and also I can count on one hand the times that there's been a difference between dev and production with vite SPAs.

Reply:
Nevermind, [your comment above](https://github.com/JRaviLab/molevolvr2.0/pull/3#discussion_r1697486205) clarifies that I was misunderstanding the purpose of this production "section" in the docker file (not to run a production build of the frontend, just to build a production version and make it accessible to a different part of the stack that will serve it).